File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/swift/F15.P1.swift
func combine() -> String {
// BEGIN PROBE F15.P1
func head<T>(_ xs: [T]) -> T { xs[0] }
let a = head([4, 5, 6])
let b = head(["p", "q"])
return String(a) + b
// END PROBE F15.P1
}

print(combine())

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/java/F15.P1.java
import java.util.List;

class Main {
    // BEGIN PROBE F15.P1
    static <T> T head(List<T> values) {
        return values.get(0);
    }

    static String combine() {
        int a = head(List.of(4, 5, 6));
        String b = head(List.of("p", "q"));
        return a + b;
    }
    // END PROBE F15.P1

    public static void main(String[] args) {
        System.out.println(combine());
    }
}
